### PR TITLE
Add stacrs

### PR DESCRIPTION
**Pull request by weiji14**
A no-dependency Python package for STAC, using Rust under the hood! Repo at https://github.com/gadomski/stacrs

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by weiji14** [comment]
Getting this build error:

```
+ cargo-bundle-licenses --format yaml --output THIRDPARTY.yml
Error: `cargo metadata` exited with an error: error: package `$SRC_DIR/crates/cli` is listed in default-members but is not a member
for workspace at $SRC_DIR/Cargo.toml.
```

I downloaded the sdist (stacrs-0.2.2.tar.gz) from https://pypi.org/project/stacrs/0.2.2/#files, and saw that the Cargo.toml at has these lines:

```
[workspace]
resolver = "2"
members = ["python"]
default-members = [
    "crates/api",
    "crates/cli",
    "crates/core",
    "crates/extensions",
]
```

These appear to be derived from https://github.com/stac-utils/stac-rs/blob/d13c0f474c1e3b2fc3854bdf60f10608b273c689/Cargo.toml#L13-L18. The 'crates/cli` folder is indeed missing in the sdist, so need to see how to handle this...

**Comment by weiji14** [comment]
Note to self: was getting this error on `openssl-sys` during compilation at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1089429&view=logs&j=4cabe686-70ae-553a-7fd0-310379f2cbac&t=6a4fc7c9-c31a-5115-eff9-6479d72b69ff&l=1143:

```
   Compiling openssl-sys v0.9.104
   Compiling futures-channel v0.3.31
   Compiling futures-macro v0.3.31
   Compiling fnv v1.0.7
   Compiling semver v1.0.23
   Compiling base64 v0.22.1
   Compiling futures-io v0.3.31
   Compiling icu_locid_transform_data v1.5.0
   Compiling pin-utils v0.1.0
   Compiling futures-task v0.3.31
   Compiling futures-util v0.3.31
   Compiling icu_locid_transform v1.5.0
   Compiling http v1.1.0
   Compiling pyo3-build-config v0.22.6
The following warnings were emitted during compilation:
warning: openssl-sys@0.9.104: configuring OpenSSL build: Command 'perl' not found. Is perl installed?
warning: openssl-sys@0.9.104: openssl-src: failed to build OpenSSL from source
error: failed to run custom build command for `openssl-sys v0.9.104`
Caused by:
  process didn't exit successfully: `$SRC_DIR/target/release/build/openssl-sys-583d03d9a9661d2c/build-script-main` (exit status: 1)
...
...
...
  running cd "$SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-03c739d734f0bbb7/out/openssl-build/build/src" && env -u CROSS_COMPILE AR="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ar" CC="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc" RANLIB="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ranlib" "perl" "./Configure" "--prefix=$SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-03c739d734f0bbb7/out/openssl-build/install" "--openssldir=/usr/local/ssl" "no-shared" "no-module" "no-ssl3" "no-tests" "no-comp" "no-zlib" "no-zlib-dynamic" "--libdir=lib" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-march=nocona" "-mtune=haswell" "-ftree-vectorize" "-fPIC" "-fstack-protector-strong" "-fno-plt" "-O2" "-ffunction-sections" "-pipe" "-isystem" "$PREFIX/include" "-fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/stacrs-0.3.0" "-fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix"
  cargo:warning=configuring OpenSSL build: Command 'perl' not found. Is perl installed?
  cargo:warning=openssl-src: failed to build OpenSSL from source
  --- stderr
  Error configuring OpenSSL build:
      Command 'perl' not found. Is perl installed?
      Command failed: cd "$SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-03c739d734f0bbb7/out/openssl-build/build/src" && env -u CROSS_COMPILE AR="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ar" CC="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-cc" RANLIB="$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ranlib" "perl" "./Configure" "--prefix=$SRC_DIR/target/x86_64-unknown-linux-gnu/release/build/openssl-sys-03c739d734f0bbb7/out/openssl-build/install" "--openssldir=/usr/local/ssl" "no-shared" "no-module" "no-ssl3" "no-tests" "no-comp" "no-zlib" "no-zlib-dynamic" "--libdir=lib" "no-md2" "no-rc5" "no-weak-ssl-ciphers" "no-camellia" "no-idea" "no-seed" "linux-x86_64" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-march=nocona" "-mtune=haswell" "-ftree-vectorize" "-fPIC" "-fstack-protector-strong" "-fno-plt" "-O2" "-ffunction-sections" "-pipe" "-isystem" "$PREFIX/include" "-fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/stacrs-0.3.0" "-fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix"
  warning: build failed, waiting for other jobs to finish...
  💥 maturin failed
    Caused by: Failed to build a native library through cargo
```

Fixed this in commit 1830fe36d1ad1e1a0ddc5941dbc2354de0cc9769 by adding the `OPENSSL_DIR` and `OPENSSL_NO_VENDOR` environment variables, following what was done at https://github.com/conda-forge/staged-recipes/pull/25709#discussion_r1527146818.

**Comment by gadomski** [comment]
@weiji14 if the duckdb dependency is being a pain, I'd be open to removing it from the Python library. It's pretty experimental right now and could be more trouble than its worth.

**Comment by weiji14** [comment]
I saw a similar issue about cstdlib mentioned at https://github.com/conda-forge/ray-packages-feedstock/pull/183#issuecomment-2435908545, and the workaround was to [downgrade to macos-12](https://github.com/conda-forge/ray-packages-feedstock/pull/183#issuecomment-2442140598), but considering that macos-12 is meant to be deprecated on Azure next month (https://github.com/conda-forge/staged-recipes/pull/27770), probably not good to downgrade from the default macos-13 to macos-12 here.

Let me try a newer `MACOSX_SDK_VERSION` - https://conda-forge.org/docs/maintainer/knowledge_base/#requiring-newer-macos-sdks, and see if it works before we think of yanking out libduckdb-sys. I'm surprised that https://github.com/duckdb/duckdb-rs/tree/v1.1.1 doesn't have CI for macOS, so don't really have a guide on what version to use (plus I don't have a macOS device). Might involve a bit of trial and error :slightly_smiling_face:

File: recipes/stacrs/recipe.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  license_file:
+    - LICENSE-APACHE
+    - LICENSE-MIT
+    - THIRDPARTY.yml
+  homepage: https://www.gadom.ski/stacrs
+  repository: https://github.com/gadomski/stacrs
+
+extra:
+  recipe-maintainers:
+    - weiji14
```
Comment (weiji14) [inline recipes/stacrs/recipe.yaml:65]: @gadomski, would you like to be a conda-forge recipe maintainer for `stacrs` (need a yes/no answer)?

```suggestion
  recipe-maintainers:
    - gadomski
    - weiji14
```

**Review by weiji14 — COMMENTED** [review]
Ok, compilation on macOS works now using `MACOSX_SDK_VERSION=10.14` :tada:

```
      Compiling stacrs v0.3.0 ($SRC_DIR)
       Finished `release` profile [optimized] target(s) in 12m 04s
   📖 Found type stub file at stacrs.pyi
   📦 Built wheel for CPython 3.12 to $SRC_DIR/target/wheels/stacrs-0.3.0-cp312-cp312-macosx_10_13_x86_64.whl
   $SRC_DIR/target/wheels/stacrs-0.3.0-cp312-cp312-macosx_10_13_x86_64.whl
   Building wheel for stacrs (pyproject.toml): finished with status 'done'
   Created wheel for stacrs: filename=stacrs-0.3.0-cp312-cp312-macosx_10_13_x86_64.whl size=16636665 sha256=2a6a47c278b33402e3282fc1853baf28a1432bfd7bda1c73da49c0be7008a3b8
   Stored in directory: /private/var/folders/p1/44pzfl0j2m301zzfb0fv0p380000gn/T/pip-ephem-wheel-cache-gb3wq8_j/wheels/01/22/fa/922df9bc9d1840c4c4b5f2153683491f3278d12edd0c165cd2
 Successfully built stacrs
 Installing collected packages: stacrs
 Successfully installed stacrs-0.3.0
```

File: recipes/stacrs/recipe.yaml (hunk 1 above)
Comment (gadomski) [reply recipes/stacrs/recipe.yaml:65]: Yup, I will, thanks for setting this up? If you have bandwidth, we could add this to the install instructions over on **stacrs** too

**Comment by weiji14** [comment]
@conda-forge/help-rust @conda-forge/help-python ready for review!

File: recipes/stacrs/recipe.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - pip
+    - python
+  run:
+    - python
+
+tests:
+  - python:
+      imports:
+        - stacrs
+  - requirements:
```
[the remark this answers is not in the document]
Comment (weiji14) [reply recipes/stacrs/recipe.yaml:47]: Hmm ok, I tried it at commit 26563d83dec080904c913a42297c072fbbf13af9 but got the following error:

```
Error:   × Failed to parse recipe

Error:   × Parsing: missing field `script`
    ╭─[48:17]
 47 │           pip_check: true
 48 │ ╭─▶   - requirements:
 49 │ │         run:
 50 │ │           - pip
 51 │ │   
 52 │ ├─▶ about:
    · ╰──── here
 53 │       summary: A no-dependency Python package for STAC, using Rust under the hood.
    ╰────
  help: expected field `script` to be a list of commands
```

Maybe I should remove the `requirements.run` part, let me try that.

**Comment by weiji14** [comment]
@conda-forge/help-rust @conda-forge/help-python, gentle ping again to get this merged if possible. Builds are passing on all platforms now.

**Review by xhochy — APPROVED** [review]

**Pull request merged**